bin/compare_api_surfaces.py

```
@@ -0,0 +1,356 @@
+            return True
+
+        if not master_param["required"] and pr_param["required"]:
+            return True
+
+        pr_index = found_index + 1
+
+    for trailing_param in pr_params[pr_index:]:
+        if trailing_param["required"]:
+            return True
+
+    return False
```

Ruff (0.15.7)

[warning] Use `return any(trailing_param["required"] for trailing_param in pr_params[pr_index:])` instead of `for` loop

Replace with `return any(trailing_param["required"] for trailing_param in pr_params[pr_index:])`

(SIM110)